modules/derivation-common/interface.nix

```
@@ -103,32 +111,32 @@
       type = t.package;
     };
 
+    depsRequired = l.mkOption {
+      type = t.attrsOf t.bool;
+      default = {};
+    };
+
     /*
-      This allows defining drvs in an encapsulated manner, while maintaining
-        the capability to depend on external attributes
+      Attrset of required dependencies.
+      For example, when `config.depsRequired.bash = true`, then this creates:
+      {
+        options.deps = l.mkOption {
+          description = "spcify a package for the dependency bash"
+          type = t.raw;
+          default = config.depsFrom.bash;
+        }
+      }
     */
-    deps = l.mkOption {
-      description = ''
-        All dependencies of the package. This option should be set by the "outer world" and can be used to inherit attributes from `pkgs` or `inputs` etc.
+    deps =
+      l.mapAttrs
+      mkDepOpt
+      (l.filterAttrs (_: enabled: enabled) config.depsRequired);
 
-        By separating the task of retrieving things from the outside world, it is ensured that the dependencies are overridable.
-        Nothing will stop users from adding `pkgs` itself as a dependency, but this will make it very hard for the user of the package to override any dependencies, because they'd have to figure out a way to insert their changes into the Nixpkgs fixpoint. By adding specific attributes to `deps` instead, the user has a realistic chance of overriding those dependencies.
-
-        So deps should be specific, but not overly specific. For instance, the caller shouldn't have to know the version of a dependency in order to override it. The name should suffice. (e.g. `nix = nixVersions.nix_2_12` instead of `inherit (nixVersions) nix_2_12`.
-      '';
-      type =
-        t.coercedTo
-        (t.functionTo (t.lazyAttrsOf t.raw))
-        callDeps
-        (t.lazyAttrsOf t.raw);
+    # Unspecified `deps` are taken from `depsFrom`. It's a source for defaults.
+    depsFrom = l.mkOption {
```

Seems like it should be this then

```suggestion
    defaultDeps = l.mkOption {
```

---

or `depsDefaults`